### PR TITLE
Feature/reducers and actions

### PR DESCRIPTION
**Pull request by Braingenius2**
# Added reducers and actions

This pull request adds Redux to the bookstore application. Redux is a state management library that helps to keep track of the application state. 

In this pull request, I implemented the following:

- [x] Installed Redux Toolkit and React Redux
- [x] Created a directory to store Redux logic
- [x] Configured a Redux store
- [x] Defined a slice of state for books
- [x] Defined a slice of state for categories
- [x] Structured the application files using a "feature folder" approach and uses the ducks pattern

### CODE REVIEW

File: src/redux/store.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { configureStore } from '@reduxjs/toolkit';
+
+const store = configureStore({
+  reducer: {},
```
Comment (zieeco) [inline src/redux/store.js]: - Hey, you've not added your reducers, please, add them. it should look something like this 
`reducer: { books: booksReducer, categories: categoriesReducer }`

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Braingenius2,





Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice reducers action ✔️
- Descriptive PR. ✔️
- Good commit messages :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions

- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/store.js (hunk 1 above)
Comment (Braingenius2) [reply src/redux/store.js]: Thanks so much. oops! Omitted them. Just made the required changes.

**Review by Strangeal — APPROVED** [review]
Hi @Braingenius2, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**